tests/unit/mathematics/linalg/ElementwiseProduct_unittest.cc

```diff
@@ -81,10 +78,7 @@ TEST(ElementwiseProduct, CGPUMatrix_eigen3_backend)
 	for (int32_t i=0; i<9; i++)
 		EXPECT_NEAR(A[i]*B[i], C[i], 1e-15);
 }
-#endif // HAVE_VIENNACL
-#endif // HAVE_EIGEN3
 
-#ifdef HAVE_VIENNACL
```

Comment:
I removed `ifdef HAVE_VIENNACL `because the structure before was like:

```
ifdef HAVE_EIGEN3
...
ifdef HAVE_VIENNACL
...
endif // HAVE_VIENNACL
endif // HAVE_EIGEN3

ifdef HAVE_VIENNACL
...
endif // HAVE_VIENNACL
```

I feel like when HAVE_EIGEN3 is removed there's no need to separate two HAVE_VIENNACL